src/components/views/settings/AvatarSetting.tsx

```diff
@@ -67,34 +118,40 @@ const AvatarSetting: React.FC<IProps> = ({ avatarUrl, avatarAltText, avatarName,
     if (uploadAvatar) {
         // insert an empty div to be the host for a css mask containing the upload.svg
         uploadAvatarBtn = (
-            <AccessibleButton
-                onClick={uploadAvatar}
-                className="mx_AvatarSetting_uploadButton"
-                aria-labelledby={a11yId.current}
-                {...hoveringProps}
-            />
+            <>
+                <AccessibleButton
+                    onClick={uploadAvatar}
+                    className="mx_AvatarSetting_uploadButton"
+                    aria-labelledby={a11yId.current}
+                />
+                <input
+                    type="file"
+                    ref={fileInputRef}
+                    className="mx_AvatarSetting_avatarUpload"
```

Comment:
it's not obvious, looking at this code, that the component will be hidden. It might be better to use an inline `style` setting? Or maybe at least a comment saying that `className: mx_AvatarSetting_avatarUpload` means it will be hidden?